### PR TITLE
[Telink] Fix simulatedIndex max number & disable NET_MGMT

### PR DESCRIPTION
**Pull request by s07641069**
**Change overview**

- Remove NET_CONFIG_SETTINGS which select NET_MGMT
- Fix SensorManager simulatedIndex max number calculation

**Testing**

Tested manually with chip-tool.

Steps:

- Run: $ chip-tool pairing ble-thread <...>
- Wait till success
- Run: $ chip-tool onoff on ${NODE_ID }1
- Wait till success
- Run: $ chip-tool onoff off ${NODE_ID }1
- Wait till success

### CODE REVIEW

**Review by andriy-bilynskyy — APPROVED** [review]

File: config/telink/app/zephyr.conf [hunk 1]
```diff
@@ -122,6 +118,9 @@ CONFIG_SETTINGS_NVS_SECTOR_SIZE_MULT=8
 # Enable NVS lookup cache
 CONFIG_NVS_LOOKUP_CACHE=y
 
+# Enable extended discovery
+CONFIG_CHIP_EXTENDED_DISCOVERY=y
```
Comment (bzbarsky-apple) [inline config/telink/app/zephyr.conf:122]: Should this be on by default for the platform?  Generally speaking, extended discovery is only useful for some kinds of devices...

Comment (s07641069) [reply config/telink/app/zephyr.conf:122]: Prefer to use by default for Telink platform as well as other vendors did.
If can be disabled by developer for any specific app if needed.

Comment (s07641069) [reply config/telink/app/zephyr.conf]: Revert this change on current PR.

File: examples/temperature-measurement-app/telink/src/SensorManager.cpp [hunk 2]
```diff
@@ -49,7 +49,7 @@ int16_t SensorManager::SensorEventHandler()
 #ifdef TEMPERATURE_SIMULATION_IS_USED
     static uint8_t nbOfRepetition = 0;
     static uint8_t simulatedIndex = 0;
-    if (simulatedIndex >= sizeof(mSimulatedTemp) - 1)
+    if (simulatedIndex >= sizeof(mSimulatedTemp) / sizeof(mSimulatedTemp[0]))
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/telink/src/SensorManager.cpp:52]: Any reason this is not using ArraySize()?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/temperature-measurement-app/telink/src/SensorManager.cpp (hunk 2 above)
Comment (s07641069) [reply examples/temperature-measurement-app/telink/src/SensorManager.cpp]: Fixed, didn't see this macro

**Review by andy31415 — APPROVED** [review]

**Pull request merged**